modules/installation-gcp-permissions.adoc

```diff
@@ -29,6 +41,294 @@ account requires the following permissions. If you deploy your cluster into an e
+[%collapsible]
+====
+* `monitoring.timeSeries.list`
+* `serviceusage.quotas.get`
+* `serviceusage.services.list`
+====
+
+.Required IAM permissions for installation
+[%collapsible]
+====
+* `iam.roles.get`
+====
+endif::ipiupi[]
+
+ifdef::ipi[]
+.Optional Images permissions for installation
+[%collapsible]
+====
+* `compute.images.list`
+====
+endif::ipi[]
+ifdef::upi[]
+.Required Images permissions for installation
+[%collapsible]
+====
+* `compute.images.list`
+====
+endif::upi[]
+
+ifdef::ipiupi[]
+.Optional permission for running gather bootstrap
+[%collapsible]
+====
+* `compute.instances.getSerialPortOutput`
+====
+
+.Required permissions for deleting network resources
+[%collapsible]
+====
+* `compute.addresses.delete`
+* `compute.addresses.deleteInternal`
+* `compute.addresses.list`
+* `compute.firewalls.delete`
+* `compute.firewalls.list`
+* `compute.forwardingRules.delete`
+* `compute.forwardingRules.list`
+* `compute.networks.delete`
+* `compute.networks.list`
+* `compute.networks.updatePolicy`
+* `compute.routers.delete`
+* `compute.routers.list`
+* `compute.routes.list`
+* `compute.subnetworks.delete`
+* `compute.subnetworks.list`
+====
+
+.Required permissions for deleting load balancer resources
+[%collapsible]
+====
+* `compute.regionBackendServices.delete`
+* `compute.regionBackendServices.list`
+* `compute.targetPools.delete`
+* `compute.targetPools.list`
+====
+
+.Required permissions for deleting DNS resources
+[%collapsible]
+====
+* `dns.changes.create`
+* `dns.managedZones.delete`
+* `dns.managedZones.get`
+* `dns.managedZones.list`
+* `dns.resourceRecordSets.delete`
+* `dns.resourceRecordSets.list`
+====
+
+.Required permissions for deleting Service Account resources
+[%collapsible]
+====
+* `iam.serviceAccounts.delete`
+* `iam.serviceAccounts.get`
+* `iam.serviceAccounts.list`
+* `resourcemanager.projects.getIamPolicy`
+* `resourcemanager.projects.setIamPolicy`
+====
+
+.Required permissions for deleting compute resources
+[%collapsible]
+====
+* `compute.disks.delete`
+* `compute.disks.list`
+* `compute.instanceGroups.delete`
+* `compute.instanceGroups.list`
+* `compute.instances.delete`
+* `compute.instances.list`
+* `compute.instances.stop`
+* `compute.machineTypes.list`
+====
+
+.Required for deleting storage resources
+[%collapsible]
+====
+* `storage.buckets.delete`
+* `storage.buckets.getIamPolicy`
+* `storage.buckets.list`
+* `storage.objects.delete`
+* `storage.objects.list`
+====
+
+.Required permissions for deleting health check resources
+[%collapsible]
+====
+* `compute.healthChecks.delete`
+* `compute.healthChecks.list`
+* `compute.httpHealthChecks.delete`
+* `compute.httpHealthChecks.list`
+====
+endif::ipiupi[]
+
+ifdef::ipi[]
+.Optional Images permissions for deletion
+[%collapsible]
+====
+* `compute.images.list`
+====
+endif::ipi[]
+
+ifdef::upi[]
+.Required permissions for deleting image resources
+[%collapsible]
+====
+* `compute.images.delete`
+* `compute.images.list`
+====
+endif::upi[]
```

Comment:
>  With this said, suggest to rename ``Optional Images permissions for deletion`` into ``Required Images permissions for deletion``. WDYT? Thanks!

@xenolinux Following changes will be required to incorporate the above.

```suggestion

.Required permissions for deleting image resources
[%collapsible]
====
* `compute.images.delete`
* `compute.images.list`
====
endif::ipiupi[]
```

Reply:
Sorry, don't quite understand. IPI only requires `compute.images.list` permission during uninstallation, the other permission `compute.images.delete` is not needed during IPI installation or uninstallation. Could you clarify a bit? Thanks in advance!

Reply:
Ok my bad. I will update the suggested changes. Thanks for pointing that out @jianli-wei .

Reply:
Updated the suggestion. @jianli-wei Let me know if the suggestion looks fine.

Reply:
Addressed

Reply:
Yes, it looks fine now. 

But, as for the `Required Images permissions` and `Required Images permissions for installation` in UPI part, I suggest to move the permissions of the former to the latter, and remove the former section, because those 4 permissions are for installation. Thanks!

Reply:
@jianli-wei Addressed

Reply:
Sorry, please help removing one duplicated `compute.images.list` from the list, thanks!

Reply:
@jianli-wei Done

Reply:
Please also sort those 4 permissions, thanks!

Reply:
@xenolinux

Reply:
@jianli-wei I believe its - `create`, `delete`, `get`, `list` - sorted in that manner

Reply:
Correct, thanks!